pkg/kube/inject/inject.go

```diff
@@ -465,10 +464,10 @@ func injectionStatus(pod *corev1.Pod) *SidecarInjectionStatus {
 
 	// default case when injected pod has explicit status
 	var iStatus SidecarInjectionStatus
-	if err := json.Unmarshal(statusBytes, &iStatus); err == nil {
-		return &iStatus
+	if err := json.Unmarshal(statusBytes, &iStatus); err != nil {
```

Comment:
What is the intent of this change?

Reply:
IMO, the go programming specification is first process error circumstance , then process normal circumstance, so do the change